### PR TITLE
Javascript Agent Documentation 

### PR DESCRIPTION
**Pull request by juan-fernandez**

### CODE REVIEW

File: docs/javascript-installation.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+```
+
+## Environment variables
+
+The following environment variables need to be configured when instrumenting your tests:
+
+| Environment variable | Default value | Description                                            |
+| -------------------- | ------------- | ------------------------------------------------------ |
+| `$SCOPE_API_KEY`     |               | API key to use when sending data to Scope              |
+| `$SCOPE_HOST`        |               | API endpoint of the Scope installation to send data to |
```
Comment (juan-fernandez) [inline docs/javascript-installation.md]: these are not following the standard but since it needs a new release of the agent, I'd rather change it later

Comment (fermayo) [reply docs/javascript-installation.md]: Can we change it and re-release the agent?

Comment (juan-fernandez) [reply docs/javascript-installation.md]: yeah I can do that

Comment (juan-fernandez) [reply docs/javascript-installation.md]: done

File: docs/javascript-installation.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+---
+id: javascript-installation
+title: Scope Javascript Agent installation
+sidebar_label: Installation
+---
+
+Installation of the Scope Agent is done via [npm](https://www.npmjs.com/package/@undefinedlabs/scope-agent).
+
+```bash
```
Comment (fermayo) [inline docs/javascript-installation.md]: We can use [docusaurus code tabs feature](https://docusaurus.io/docs/en/doc-markdown#language-specific-code-tabs) here

File: docs/javascript-installation.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+Where `cypress/support/index.js`:
+
+```javascript
+import { initializeCypress } from "@undefinedlabs/scope-agent";
+
+initializeCypress();
+
+... // any other configuration here
```
Comment (fermayo) [inline docs/javascript-installation.md]:
```suggestion
// ... any other configuration here
```

File: docs/javascript-installation.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+Then you have to configure [jest reporter](https://jestjs.io/docs/en/configuration#reporters-array-modulename-modulename-options)
+to point to this file.
+
+For example:
+
+In your `jest.config.js`:
+
+```javascript
+module.exports = {
+  ...
```
Comment (fermayo) [inline docs/javascript-installation.md]:
```suggestion
  // ...
```

File: docs/javascript-installation.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  reporters: ["<rootDir>/reporter.js"]
+  ...
```
Comment (fermayo) [inline docs/javascript-installation.md]:
```suggestion
  // ...
```

File: docs/javascript-installation.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+#### Note for Create React App users
+
+CRA does not allow `reporter` configuration for jest yet (more info in https://github.com/facebook/create-react-app/issues/2474) so you may not use the `jest` field in your `package.json`.
+
+One alternative is running your test with inline configuration:
+
+In your `package.json`:
+
+```
```
Comment (fermayo) [inline docs/javascript-installation.md]:
```suggestion
```json
```

I'd basically try to make the syntax be valid json syntax

File: docs/javascript-installation.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+const { initCypressPlugin } = require('@undefinedlabs/scope-agent')
+
+module.exports = async (on, config) => {
+  ... // any other plugin goes here
+  const newConfig = await initCypressPlugin(on, config)
+  return newConfig
+}
+```
+
+## Environment variables
```
Comment (fermayo) [inline docs/javascript-installation.md]: Do these work even in CRA? No need to prefix them as `REACT_APP_`?

Comment (juan-fernandez) [reply docs/javascript-installation.md]: yeah they do. I believe `REACT_APP_` is just for environment variables consumed by the bundle, but we are consuming them in jest and cypress processes

File: docs/javascript-installation.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+to point to this file.
+
+For example:
+
+In your `jest.config.js`:
+
+```javascript
+module.exports = {
+  // ...
+  reporters: ["<rootDir>/reporter.js"]
```
Comment (javidalpe) [inline docs/javascript-installation.md]: You are using `src/reporter.js` as example path, but now you're using `<rootDir>/reporter.js`. It should be src/reporter.js again, right?

File: docs/javascript-installation.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+  // ...
+};
+```
+
+#### Note for Create React App users
```
Comment (javidalpe) [inline docs/javascript-installation.md]: I'd say `Create React App` and give the command line first, a then the explanation. I don't care if CRA allows or not reporter config in the package.json

File: docs/javascript-installation.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+CRA does not allow `reporter` configuration for jest yet (more info in https://github.com/facebook/create-react-app/issues/2474) so you may not use the `jest` field in your `package.json`.
+
+One alternative is running your test with inline configuration:
+
+In your `package.json`:
+
+```json
+...
+"scripts": {
+  "test:reporters": "react-scripts test --reporters=default --reporters=./src/reporter.js"
```
Comment (javidalpe) [inline docs/javascript-installation.md]: It should be more semantic, like: `test:scope` or `test:tracing` or even `test:debug`

File: docs/javascript-installation.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+}
+...
+```
+
+### Cypress tests
+
+If you want to instrument tests run by Cypress you need to add the scope-agent plugin.
+
+An example [cypress.json](https://docs.cypress.io/guides/references/configuration.html#Folders-Files):
```
Comment (javidalpe) [inline docs/javascript-installation.md]:
```suggestion
Add the following to [cypress.json](https://docs.cypress.io/guides/references/configuration.html#Folders-Files):
```

File: docs/javascript-installation.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+const { initCypressPlugin } = require("@undefinedlabs/scope-agent");
+
+module.exports = async (on, config) => {
+  // ... any other plugin goes here
+  const newConfig = await initCypressPlugin(on, config);
+  return newConfig;
+};
+```
+
+## Environment variables
```
Comment (javidalpe) [inline docs/javascript-installation.md]: I think we should put this section as `## Prerequisites` before anything else.

Comment (juan-fernandez) [reply docs/javascript-installation.md]: while I would agree, this is the standard for the rest of the agents so I prefer to keep it this way

File: docs/javascript-installation.md (hunk 8 above)
Comment (juan-fernandez) [reply docs/javascript-installation.md]: you need to specify `<rootDir>` in `jest.config.js` https://jestjs.io/docs/en/configuration#rootdir-string. But there was indeed a mistake. It should be `<rootDir>/src/reporter.js` to keep consistency

**Pull request merged**